### ISSUE TITLE
Poor editing performance with long filter expression

### ISSUE THREAD

**Issue by CryogenicPotato** [bcolpaert/Colso.Xrm.DataTransporter#22]
Putting a long FetchXML condition such as:

```
<link-entity name="opportunity" to="objectid" from="opportunityid" alias="ai">
  <filter type="or">
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
	<condition value="{394E121B-1AAC-E611-80E7-005056A60883}" attribute="parentaccountid" uitype="account" uiname="xxxxxxxx" operator="eq" />
  </filter>
</link-entity>
```

into Filter area causes that when clicking on Filter button a second time to edit the filter expression, the application hangs for a long time before opening the filter window.

**Comment by bcolpaert** [comment]
I'll have a look at it and try to fix this with the next release

**Comment by bcolpaert** [comment]
This issue is fixed in version 1.2018.7.1
